src/sentry/api/endpoints/user_social_identity_details.py

```diff
@@ -29,20 +27,25 @@ def delete(self, request, user, identity_id):
         except UserSocialAuth.DoesNotExist:
             return Response(status=404)
 
-        backend = get_backend(auth.provider, request, "/")
-        if backend is None:
-            raise Exception(u"Backend was not found for request: {}".format(auth.provider))
-
-        # stop this from bubbling up errors to social-auth's middleware
-        # XXX(dcramer): IM SO MAD ABOUT THIS
-        try:
-            backend.disconnect(user, identity_id)
-        except Exception as exc:
-            import sys
-
-            exc_tb = sys.exc_info()[2]
-            six.reraise(Exception, exc, exc_tb)
-            del exc_tb
+        # HACK(joshuarli): I've tried every method i could come up with to call social
+        # auth's disconnect, but for reasons i fail to understand, the backend object
+        # (e.g. AsanaOAuth2) is strange while it's going through social auth's disconnect
+        # code. For example, you might run into it not having the "strategy" attribute,
+        # which I hacked around with:
+        #
+        #   class AsanaOAuth2(_AsanaOAuth2):
+        #   +     strategy = load_strategy()
+        #
+        # But then you'll likely encounter:
+        # unbound method setting() must be called with AsanaOAuth2 instance as first argument
+        # (got str instance instead)
+        #
+        # And then I haven't been able to figure out how to get past that.
+        # What essentially needs to happen is that the UserSocialAuth needs to be deleted,
+        # which is IMO a better hack than improperly using social auth disconnect.
+        # The only thing we'll lose from this approach is SOCIAL_AUTH_REVOKE_TOKENS_ON_DISCONNECT,
+        # but I don't see that being much of an issue as tokens (at least for asana) expire after an hour.
+        auth.delete()
```

Comment:
@joshuarli what happens with this? Does it just delete the row in our DB?

Reply:
Yes, it deletes the appropriate row in the social_auth_usersocialauth table. That's fundamentally what it means to dissociate an identity.